Tests/CLICommands/Steps/DiscoverSourceFilesSpec.swift

```diff
@@ -11,94 +11,191 @@ class DiscoverSourceFilesSpec: QuickSpec {
                         state.tempDirectoryURL = URL(fileURLWithPath: path, isDirectory: true)
-                        
+
                         result = discoverSourceFiles.run(with: state)
                     }
-                    
+
                     it("returns the discovered source files, sorted alphabetically, with the exclude list applied") {
                         guard case .success(let stateChanges) = result! else {
                             fail("expected success but got \(String(describing: result!))")
                             return
                         }
-                        
+
                         expect(stateChanges) == [.sourceFileCandidatesDiscovered([
                             "\(path)/Directory1/file3.swift",
                             "\(path)/Directory2/Directory3/file6.swift",
                             "\(path)/file1.swift",
                             "\(path)/file2.swift",
-                            ])]
+                        ])]
                     }
                 }
             }
-            
+
             context("when it doesn't discover any Swift files in a project directory") {
                 var result: Result<[RunCommandState.Change], MuterError>! // keep this as locally defined as possible to avoid test pollution
-                
+
                 context("because a project directory contains 0 Swift files") {
                     beforeEach {
                         state.tempDirectoryURL = URL(fileURLWithPath: "\(self.fixturesDirectory)/FilesToDiscover/Directory4",
                                                      isDirectory: true)
                         result = discoverSourceFiles.run(with: state)
                     }
-                    
+
                     it("cascades a failure") {
                         guard case .failure(.noSourceFilesDiscovered) = result! else {
                             fail("expected noSourceFilesDiscovered but got \(String(describing: result!))")
                             return
-                            
                         }
                     }
                 }
-                
+
                 context("because the directory doesn't exist") {
                     beforeEach {
                         let path = "I don't exist"
+
+                        state.tempDirectoryURL = URL(fileURLWithPath: path, isDirectory: true)
+                        result = discoverSourceFiles.run(with: state)
+                    }
+
+                    it("cascades a failure") {
+                        guard case .failure(.noSourceFilesDiscovered) = result! else {
+                            fail("expected noSourceFilesDiscovered but got \(String(describing: result!))")
+                            return
+
+                        }
+                    }
+                }
+            }
+
+            context("when there is a list of files to mutate") {
```

Comment:
now that i think of it, we don't have any tests that prove we have relative pathing working. does your change add support for that? if so, it's valuable to have a test for that

Reply:
I think it should, `Pathos` probably is smart enough to handle it

Reply:
Drive it out with a test. :)